StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/Link/LinkURLGeneratorTests.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  LinkURLGeneratorTests.swift
+//  StripePaymentSheetTests
+//
+
+import Foundation
+
+@_spi(STP) @testable import StripeCore
+@_spi(STP) @testable import StripePayments
+@_spi(STP) @_spi(ExperimentalPaymentSheetDecouplingAPI) @testable import StripePaymentSheet
+
+import OHHTTPStubs
+import OHHTTPStubsSwift
+import StripeCoreTestUtils
+import XCTest
+
+class LinkURLGeneratorTests: XCTestCase {
+    let testParams = LinkURLParams(paymentObject: .link_payment_method,
+                                   publishableKey: "pk_test_123",
+                                   paymentUserAgent: "test",
+                                   merchantInfo: LinkURLParams.MerchantInfo(businessName: "Test test", country: "US"),
+                                   customerInfo: LinkURLParams.CustomerInfo(country: "US", email: "test@example.com"),
+                                   paymentInfo: LinkURLParams.PaymentInfo(currency: "USD", amount: 100),
+                                   experiments: [:],
+                                   flags: [:],
+                                   loggerMetadata: [:],
+                                   locale: Locale.init(identifier: "en_US").toLanguageTag())
+
+    func testURLCreation() {
+        let url = try! LinkURLGenerator.url(params: testParams)
+        XCTAssertEqual(url.absoluteString, "https://checkout.link.com/#eyJsb2dnZXJNZXRhZGF0YSI6e30sImN1c3RvbWVySW5mbyI6eyJjb3VudHJ5IjoiVVMiLCJlbWFpbCI6InRlc3RAZXhhbXBsZS5jb20ifSwicGF5bWVudE9iamVjdCI6ImxpbmtfcGF5bWVudF9tZXRob2QiLCJleHBlcmltZW50cyI6e30sInBheW1lbnRVc2VyQWdlbnQiOiJ0ZXN0IiwibG9jYWxlIjoiZW4tVVMiLCJwYXRoIjoibW9iaWxlX3BheSIsInBheW1lbnRJbmZvIjp7ImN1cnJlbmN5IjoiVVNEIiwiYW1vdW50IjoxMDB9LCJtZXJjaGFudEluZm8iOnsiY291bnRyeSI6IlVTIiwiYnVzaW5lc3NOYW1lIjoiVGVzdCB0ZXN0In0sInB1Ymxpc2hhYmxlS2V5IjoicGtfdGVzdF8xMjMiLCJmbGFncyI6e30sImludGVncmF0aW9uVHlwZSI6Im1vYmlsZSJ9")
+    }
+
+    func testURLCreationRegularUnicode() {
+        var params = testParams
+        params.customerInfo.email = "유니코드"
+        _ = try! LinkURLGenerator.url(params: params)
+        // Just make sure it doesn't fail
```

Comment:
Rather than failing, it's probably safer to hardcode the base64 encoding in this test.

Reply:
I tried that but it was annoying to update two URLs every time we change parameters, which will be pretty often during development. I kinda trust that the Unicode pipeline works, I don't know how necessary this test is tbh